tutorials/tts/MixerTTS_FastPitch_Training.ipynb

```diff
@@ -0,0 +1,693 @@
+    "    phoneme_dict=\"tts_dataset_files/cmudict-0.7b_nv22.01\",\n",
+    "    heteronyms=\"tts_dataset_files/heteronyms-030921\"\n",
+    ")\n",
+    "\n",
+    "# text tokenizer\n",
+    "text_tokenizer = EnglishPhonemesTokenizer(\n",
+    "    punct=True,\n",
+    "    stresses=True,\n",
+    "    chars=True,\n",
+    "    apostrophe=True,\n",
+    "    pad_with_space=True,\n",
+    "    g2p=g2p,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b677d015",
+   "metadata": {},
+   "source": [
+    "To accelerate and stabilize our training, we also need to extract pitch for every audio, estimate pitch statistics (mean and std) and pre-calculate alignment prior matrices. To do this, all we need to do is iterate over our data one time."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9ae062ee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def pre_calculate_supplementary_data(sup_data_path, sup_data_types, text_tokenizer, text_normalizer, text_normalizer_call_kwargs):\n",
+    "    # init train and val dataloaders\n",
+    "    stages = [\"train\", \"val\"]\n",
+    "    stage2dl = {}\n",
+    "    for stage in stages:\n",
+    "        ds = TTSDataset(\n",
+    "            manifest_filepath=f\"tests/data/asr/an4_{stage}.json\",\n",
+    "            sample_rate=16000,\n",
+    "            sup_data_path=sup_data_path,\n",
+    "            sup_data_types=sup_data_types,\n",
+    "            n_fft=1024,\n",
+    "            win_length=1024,\n",
+    "            hop_length=256,\n",
+    "            window=\"hann\",\n",
+    "            n_mels=80,\n",
+    "            lowfreq=0,\n",
+    "            highfreq=8000,\n",
+    "            text_tokenizer=text_tokenizer,\n",
+    "            text_normalizer=text_normalizer,\n",
+    "            text_normalizer_call_kwargs=text_normalizer_call_kwargs\n",
+    "\n",
+    "        ) \n",
+    "        stage2dl[stage] = torch.utils.data.DataLoader(ds, batch_size=1, collate_fn=ds._collate_fn, num_workers=1)\n",
+    "\n",
+    "    # iteration over dataloaders\n",
+    "    pitch_mean, pitch_std, pitch_min, pitch_max = None, None, None, None\n",
+    "    for stage, dl in stage2dl.items():\n",
+    "        pitch_list = []\n",
+    "        for batch in tqdm(dl, total=len(dl)):\n",
+    "            tokens, tokens_lengths, audios, audio_lengths, attn_prior, pitches, pitches_lengths = batch\n",
+    "            pitch = pitches.squeeze(0)\n",
+    "            pitch_list.append(pitch[pitch != 0])\n",
+    "\n",
+    "        if stage == \"train\":\n",
+    "            pitch_tensor = torch.cat(pitch_list)\n",
+    "            pitch_mean, pitch_std = pitch_tensor.mean().item(), pitch_tensor.std().item()\n",
+    "            pitch_min, pitch_max = pitch_tensor.min().item(), pitch_tensor.max().item()\n",
+    "            \n",
+    "    return pitch_mean, pitch_std, pitch_min, pitch_max"
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
Could you add a quick description for this code block? e.g. to tell users to modify "sup_data_folder" to the directory in which they want the supplementary data to be saved, like in the later FastPitch section.